pkg/controller/operators/catalog/operator.go

```diff
@@ -1726,22 +1726,34 @@ func (o *Operator) ExecutePlan(plan *v1alpha1.InstallPlan) error {
 					Resource: r.Name,
 				}
 
-				// Set up the dynamic client ResourceInterface
+				owner := &v1alpha1.ClusterServiceVersion{}
+				owner.SetNamespace(plan.GetNamespace())
+				owner.SetName(step.Resolving)
+
+				// Set up the dynamic client ResourceInterface and set ownerrefs
 				var resourceInterface dynamic.ResourceInterface
 				if r.Namespaced {
-					ownerutil.AddOwner(unstructuredObject, plan, false, false)
 					unstructuredObject.SetNamespace(namespace)
 					resourceInterface = o.dynamicClient.Resource(gvr).Namespace(namespace)
+
+					// Set OwnerReferences for namespace-scoped resource
+					ownerutil.AddNonBlockingOwner(unstructuredObject, owner)
+
+					// Update UIDs on all CSV OwnerReferences
+					updated, err := o.getUpdatedOwnerReferences(unstructuredObject.GetOwnerReferences(), plan.Namespace)
+					if err != nil {
+						return errorwrap.Wrapf(err, "error generating ownerrefs for unstructured object: %s", unstructuredObject.GetName())
+					}
+
+					unstructuredObject.SetOwnerReferences(updated)
 				} else {
 					resourceInterface = o.dynamicClient.Resource(gvr)
-				}
 
-				// Update UIDs on all CSV OwnerReferences
-				updated, err := o.getUpdatedOwnerReferences(unstructuredObject.GetOwnerReferences(), plan.Namespace)
-				if err != nil {
-					return errorwrap.Wrapf(err, "error generating ownerrefs for unstructured object: %s", unstructuredObject.GetName())
+					// Add owner labels to cluster-scoped resource
```

Comment:
The CSV is namespaced - should we add it as an owner to a cluster scopped resource? This doesn't seem to be a good call based on the note in [1] - are OwnerLabels an OLM construct that we use to search for Cluster Scoped resources when OLM deletes a namespaced resource?

[1] https://kubernetes.io/docs/concepts/workloads/controllers/garbage-collection/#owners-and-dependents

Reply:
These are labels, not OwnerReferences, so I'd say yes.